brain/python/circuit.cpp

```diff
@@ -0,0 +1,177 @@
+}
+
+GET_CIRCUIT_PROPERTY_FOR_GIDS(MorphologyTypes)
+GET_CIRCUIT_PROPERTY_FOR_GIDS(ElectrophysiologyTypes)
+
+object Circuit_getMorphologyURIs( const Circuit& circuit, object gids )
+{
+    return toPythonList( circuit.getMorphologyURIs( gidsFromPython( gids )));
+}
+
+#define GET_CIRCUIT_PROPERTY_VALUES(property) \
+    list Circuit_get##property( const Circuit& circuit ) \
+{                                                        \
+    return toPythonList( circuit.get##property( ));      \
+}
+GET_CIRCUIT_PROPERTY_VALUES(MorphologyNames)
+GET_CIRCUIT_PROPERTY_VALUES(ElectrophysiologyNames)
+
+list Circuit_loadMorphologies( const Circuit& circuit,
+                               object gids, Circuit::Coordinates coords )
+{
+    return toPythonList(
+        circuit.loadMorphologies( gidsFromPython( gids ), coords ));
+}
+
+object Circuit_getPositions( const Circuit& circuit, object gids )
+{
+    return toNumpy( circuit.getPositions( gidsFromPython( gids )));
+}
+
+object Circuit_getTransforms( const Circuit& circuit, object gids )
+{
+    object matrices = toNumpy( circuit.getTransforms( gidsFromPython( gids )));
+    // We want the result to be indexed using regular mathematical notation
+    // even if the actual storage is column-major.
+    return matrices.attr( "transpose" )( 0, 2, 1 );
+}
+
+object Circuit_getRotations( const Circuit& circuit, object gids )
+{
+    return toNumpy( circuit.getRotations( gidsFromPython( gids )));
+}
+
+SynapsesWrapper Circuit_getAfferentSynapses(
+    const CircuitPtr& circuit, object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getAfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getEfferentSynapses(
+    const CircuitPtr& circuit, object gids,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getEfferentSynapses( gidsFromPython( gids ),
+                                                          prefetch ),
+                            circuit );
+}
+
+SynapsesWrapper Circuit_getProjectedSynapses(
+    const CircuitPtr& circuit, object pre, object post,
+    const brain::SynapsePrefetch prefetch )
+{
+    return SynapsesWrapper( circuit->getProjectedSynapses(
+                                gidsFromPython( pre ), gidsFromPython( post ),
+                                prefetch ),
+                            circuit );
+}
+
+}
+
+void export_Circuit()
+{
+
+class_< Circuit, boost::noncopyable, CircuitPtr >
+    circuitWrapper( "Circuit", no_init );
+
+scope circuitScope = circuitWrapper;
+
+enum_< Circuit::Coordinates >( "Coordinates" )
+    .value( "global_", Circuit::Coordinates::global )
+    .value( "local", Circuit::Coordinates::local )
+    ;
+
+circuitWrapper
+    .def( "__init__", make_constructor( createCircuitFromURI ))
```

Comment:
we should add/find people which review the python API. For instance, personally, having at least an alias name mtypes() for morphology_types() seems desirable in terms of typing, readability, beauty, what not. There are other preferences or styles by other more pythonic people most probably.